src/Symfony/Bundle/SecurityBundle/Resources/config/security.php

```diff
@@ -0,0 +1,275 @@
+use Symfony\Component\Security\Http\Authentication\AuthenticationUtils;
+use Symfony\Component\Security\Http\Controller\UserValueResolver;
+use Symfony\Component\Security\Http\Firewall;
+use Symfony\Component\Security\Http\HttpUtils;
+use Symfony\Component\Security\Http\Logout\LogoutUrlGenerator;
+use Symfony\Component\Security\Http\Session\SessionAuthenticationStrategy;
+use Symfony\Component\Security\Http\Session\SessionAuthenticationStrategyInterface;
+
+return static function (ContainerConfigurator $container) {
+    $container->parameters()
+        ->set('security.role_hierarchy.roles', [])
+    ;
+
+    $container->services()
+        ->set('security.authorization_checker', AuthorizationChecker::class)
+            ->public()
+            ->args([
+                service('security.token_storage'),
+                service('security.authentication.manager'),
+                service('security.access.decision_manager'),
+                param('security.access.always_authenticate_before_granting'),
+            ])
+        ->alias(AuthorizationCheckerInterface::class, 'security.authorization_checker')
+
+        ->set('security.token_storage', UsageTrackingTokenStorage::class)
+            ->public()
+            ->args([
+                service('security.untracked_token_storage'),
+                service_locator([
+                    'session' => service('session'),
+                ]),
+            ])
+            ->tag('kernel.reset', ['method' => 'disableUsageTracking'])
+            ->tag('kernel.reset', ['method' => 'setToken'])
+        ->alias(TokenStorageInterface::class, 'security.token_storage')
+
+        ->set('security.untracked_token_storage', TokenStorage::class)
+
+        ->set('security.helper', Security::class)
+            ->args([service_locator([
+                'security.token_storage' => service('security.token_storage'),
+                'security.authorization_checker' => service('security.authorization_checker'),
+            ])])
+        ->alias(Security::class, 'security.helper')
+
+        ->set('security.user_value_resolver', UserValueResolver::class)
+            ->args([
+                service('security.token_storage'),
+            ])
+            ->tag('controller.argument_value_resolver', ['priority' => 40])
+
+        // Authentication related services
+        ->set('security.authentication.trust_resolver', AuthenticationTrustResolver::class)
+
+        ->set('security.authentication.session_strategy', SessionAuthenticationStrategy::class)
+            ->args([param('security.authentication.session_strategy.strategy')])
+        ->alias(SessionAuthenticationStrategyInterface::class, 'security.authentication.session_strategy')
+
+        ->set('security.authentication.session_strategy_noop', SessionAuthenticationStrategy::class)
+            ->args(['none'])
+
+        ->set('security.encoder_factory.generic', EncoderFactory::class)
+            ->args([[]])
```

Comment:
could be an abstract_arg as a factory without encoders (empty array) is not useful